grakn-engine/src/main/java/ai/grakn/engine/task/BackgroundTaskRunner.java

```diff
@@ -54,7 +59,15 @@ public BackgroundTaskRunner(GraknConfig graknConfig) {
     public void register(BackgroundTask backgroundTask){
         if(!registeredTasks.contains(backgroundTask)) {
             registeredTasks.add(backgroundTask);
-            threadPool.scheduleAtFixedRate(backgroundTask::run, backgroundTask.period(), backgroundTask.period(), TimeUnit.SECONDS);
+            threadPool.scheduleAtFixedRate(() -> {
+                try {
+                    backgroundTask.run();
+                } catch (Throwable e) {
```

Comment:
Pay attention to how I deliberately catching every kind of exceptions / errors so that they get logged.